app/views/pages/dashboard.html.erb

```diff
@@ -72,24 +72,39 @@
+            <% if c.challenge_type == "KM" %>
+          <% header = "#{c.target_distance} km from" %>
+          <% else %>
+          <% header = "#{c.target_sessions} sessions from" %>
+          <% end %>
+            <p><%= c.activity_type.capitalize %> <%= header %> <%= c.start_date.strftime("%d %b") %> to <%= c.end_date.strftime("%d %b %Y") %></p>
             <div class="challenge-details mt-3">
               <div class="challenge-detail-block pl-2 text-left">
                 <p class="mb-2">Status: <span class="badge"><%= c.status.capitalize %></span></p>
                 <p class="mb-2">Progress: <span class="badge"><%= c.progress_text %></span></p>
-                <p class="mb-2">Distance left: <%= c.target_distance - c.all_activities_km %>km</p>
+                <% if c.challenge_type == "KM"  %>
+                <p class="mb-2">Distance left: <%= c.target_distance - c.all_activities_km %></p>
+                <% else %>
+                <p class="mb-2">Sessions left: <%= c.target_sessions - c.all_activities_sessions %></p>
+                <% end %>
               </div>
               <div class="progress-block challenge-detail-block">
                 <p style="text-align: left;">Elapsed time</p>
                 <div class="progress mb-2">
                   <% a = 0 %>
-                  <% a = number_to_percentage((Date.today - c.start_date).fdiv(c.end_date - c.start_date)*100,precision: 2) if Date.today >= c.start_date%>
+                  <% a = number_to_percentage((Date.today - c.start_date).fdiv(c.end_date - c.start_date)*100,precision: 2) if Date.today >= c.start_date %>
                   <% a %>
-                  <div class="progress-bar bg-green" role="progressbar" style="width: <%= a %>;" aria-valuenow="25" aria-valuemin="0" aria-valuemax="100"><%= a %></div>
+                  <% b = a.to_i %>
+                  <% d = 100 %>
+                  <div class="progress-bar bg-green" role="progressbar" style="width: <%= a %>;" aria-valuenow="25" aria-valuemin="0" aria-valuemax="100"><%= [b, d].min %>%</div>
                 </div>
+                <% if c.challenge_type == "KM" %>
                 <p style="text-align: left;">Distance achieved</p>
                 <% distance_progress = number_to_percentage((c.all_activities_km / c.target_distance)*100, precision: 2) %>
-                <div class="progress ">
+                <% else %>
+                <p style="text-align: left;">Sessions achieved</p>
+                <% distance_progress = number_to_percentage((c.all_activities_sessions / c.target_sessions)*100, precision: 2) %>
+                <% end %>
                   <div class="progress-bar bg-green" role="progressbar" style="width: <%= distance_progress %>;" aria-valuenow="25" aria-valuemin="0" aria-valuemax="100"><%= distance_progress %></div>
                 </div>
               </div>
```

Comment:
Updates dashboard to take into account different challenges logic and rephrased the text